src/SprykerEco/Zed/Unzer/Business/Import/UnzerPaymentMethodImporter.php

```diff
@@ -88,4 +108,54 @@ public function performPaymentMethodsImport(UnzerKeypairTransfer $unzerKeypairTr
 
         $this->paymentImportAdapter->createPaymentMethods($paymentProviderTransfer, $paymentMethodTransfers);
     }
+
+    /**
+     * @param \Generated\Shared\Transfer\PaymentMethodsTransfer $paymentMethodsTransfer
+     * @param \Generated\Shared\Transfer\PaymentMethodsTransfer $childPaymentMethodsTransfer
+     *
+     * @return \Generated\Shared\Transfer\PaymentMethodsTransfer
+     */
+    protected function appendChildPaymentMethodsTransfer(
+        PaymentMethodsTransfer $paymentMethodsTransfer,
+        PaymentMethodsTransfer $childPaymentMethodsTransfer
+    ): PaymentMethodsTransfer {
+        $unzerPaymentMethodKeys = $this->getPaymentMethodKeys($paymentMethodsTransfer);
+
+        foreach ($childPaymentMethodsTransfer->getMethods() as $paymentMethodTransfer) {
+            if (!in_array($paymentMethodTransfer->getPaymentMethodKeyOrFail(), $unzerPaymentMethodKeys, true)) {
+                $paymentMethodsTransfer->getMethods()->append($paymentMethodTransfer);
+            }
+        }
+
+        return $paymentMethodsTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UnzerKeypairTransfer $unzerKeypairTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerCredentialsCollectionTransfer
+     */
+    protected function getChildUnzerCredentialsCollectionTransfer(
+        UnzerKeypairTransfer $unzerKeypairTransfer
+    ): UnzerCredentialsCollectionTransfer {
+        $unzerCredentialsConditionsTransfer = (new UnzerCredentialsConditionsTransfer())->addParentId($unzerKeypairTransfer->getIdUnzerCredentialsOrFail());
+        $unzerCredentialsCriteriaTransfer = (new UnzerCredentialsCriteriaTransfer())->setUnzerCredentialsConditions($unzerCredentialsConditionsTransfer);
+
+        return $this->unzerReader->getUnzerCredentialsCollectionByCriteria($unzerCredentialsCriteriaTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PaymentMethodsTransfer $paymentMethodsTransfer
+     *
+     * @return array<string>
+     */
+    protected function getPaymentMethodKeys(PaymentMethodsTransfer $paymentMethodsTransfer): array
+    {
+        $paymentMethodKeys = [];
+        foreach ($paymentMethodsTransfer->getMethods() as $paymentMethodTransfer) {
+            $paymentMethodKeys[] = $paymentMethodTransfer->getPaymentMethodKeyOrFail();
```

Comment:
make sure you did not introduce new `orFail` for existing API

Reply:
In this case it is generally not possible that there is no key.